File: spring-mvc-plain-java-example/src/main/java/de/cfranzen/examples/webflux/groups/CustomerGroupClient.java
package de.cfranzen.examples.webflux.groups;

import org.springframework.stereotype.Component;
import org.springframework.web.client.RestClient;

@Component
public class CustomerGroupClient {

    private final RestClient restClient;

    public CustomerGroupClient(final RestClient restClient) {
        this.restClient = restClient;
    }

    public CustomerGroup getCustomerGroup(String name) {
        return restClient
                .get()
                .uri(uriBuilder -> uriBuilder
                        .path("/api/customerGroup/{name}")
                        .build(name))
                .retrieve()
                .body(CustomerGroup.class);
    }
}

File: spring-webflux-kotlin-example/src/main/kotlin/de/cfranzen/examples/webflux/kotlin/groups/CustomerGroupClient.kt
package de.cfranzen.examples.webflux.kotlin.groups

import kotlinx.coroutines.flow.single
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient
import org.springframework.web.reactive.function.client.bodyToFlow
import org.springframework.web.util.UriBuilder

@Component
class CustomerGroupClient(
    private val webClient: WebClient
) {

    suspend fun getCustomerGroup(name: String): CustomerGroup {
        return webClient
            .get()
            .uri { uriBuilder: UriBuilder ->
                uriBuilder
                    .path("/api/customerGroup/{name}")
                    .build(name)
            }
            .retrieve()
            .bodyToFlow<CustomerGroup>()
            .single()
    }
}
